DatadogInternal/Sources/Models/SessionReplay/PrivacyLevel+SessionReplay.swift

```diff
@@ -0,0 +1,81 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+import UIKit
+
+// MARK: - DatadogExtension for UIView
+
+/// Extension on `DatadogExtension` to provide access to `SessionReplayOverride` for any `UIView`.
+extension DatadogExtension where ExtendedType: UIView {
+    /// Provides access to Session Replay override settings for the view.
+    /// Usage: `myView.dd.sessionReplayOverride.textAndInputPrivacy = .maskNone`.
+    public var sessionReplayOverride: SessionReplayOverrideExtension<ExtendedType> {
+        get {
+            return SessionReplayOverrideExtension(self.type)
+        }
+        set {}
+    }
+}
+
+// MARK: - Associated Keys
+
+private var associatedTextAndInputPrivacyKey: UInt8 = 3
+private var associatedImagePrivacyKey: UInt8 = 4
+private var associatedTouchPrivacyKey: UInt8 = 5
+private var associatedHiddenPrivacyKey: UInt8 = 6
+
+// MARK: - SessionReplayOverrideExtension
+
+/// `UIView` extension  to manage the Session Replay privacy override settings.
+public struct SessionReplayOverrideExtension<ExtendedType> {
+    private let view: ExtendedType
+
+    public init(_ view: ExtendedType) {
+        self.view = view
+    }
+
+    /// Text and input privacy override (e.g., mask or unmask specific text fields, labels, etc.).
+    public var textAndInputPrivacy: TextAndInputPrivacyLevel? {
+        get {
+            return objc_getAssociatedObject(view as AnyObject, &associatedTextAndInputPrivacyKey) as? TextAndInputPrivacyLevel
+        }
+        set {
+            objc_setAssociatedObject(view as AnyObject, &associatedTextAndInputPrivacyKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```

Comment:
/question

as per public docs of `OBJC_ASSOCIATION_RETAIN_NONATOMIC` 

https://developer.apple.com/documentation/objectivec/objc_associationpolicy/objc_association_retain_nonatomic
>Specifies a strong reference to the associated object, and that the association is not made atomically.

This means any strong reference to these extension properties will not allow deinit of views which could be potential issue for customers apps. We would like to stay away from the view life cycle and let the customer decide it.

Reply:
That's a valid point, I can probably use `OBJC_ASSOCIATION_COPY_NONATOMIC` instead.

Reply:
I've looked into it, and since the properties being associated are value types, they are copied by default, so using `OBJC_ASSOCIATION_RETAIN_NONATOMIC` or `OBJC_ASSOCIATION_COPY_NONATOMIC` will result in the same behavior. That said, switching to `OBJC_ASSOCIATION_COPY_NONATOMIC` can make sense because the overrides are meant to be isolated per view, not shared between views. This is safer in case we later decide to use reference types.

Reply:
Just to clarify: using `OBJC_ASSOCIATION_RETAIN_NONATOMIC` will retain the associated value, not the view. But in this case, using `COPY` is more explicit 👍